share-it 3.0

KnSwee opened the pull request:
Реализованы требования ТЗ 16 спринта

KnSwee committed:
share-it 3.0
Реализованы требования ТЗ 16 спринта

avfyodorov commented on gateway/src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -0,0 +1,65 @@
+
+    @GetMapping("/search")
+    public ResponseEntity<Object> getItemsToBook(@RequestHeader("X-Sharer-User-Id") Long userId,
+                                                 @RequestParam("text") String text) {
```
Лучше было бы прямо здесь проверить входной параметр text. И, если текст запроса пустой, или недопустимый, например, состоит из пробелов, то прямо отсюда и вернуть результат, не обращаться к серверу. Вернуть можно, например, пустую коллекцию.

KnSwee replied:
то есть простую проверку ифом? не валидацию?

avfyodorov replied:
Можно так :
```java
if (text == null || text.isBlank()) {
			return (ResponseEntity<Object>) Collections.emptyList();
}
```

avfyodorov commented on gateway/src/main/java/ru/practicum/shareit/ShareItGateway.java:
```diff
@@ -0,0 +1,12 @@
+@SpringBootApplication
+public class ShareItGateway {
+    public static void main(String[] args) {
+        SpringApplication.run(ShareItGateway.class, args);
```
Теперь есть (должно быть) два обработчика ошибок: один- в модуле гейтвей,  а второй-это сервер. И ошибки перехватываются  и обрабатываются, соответственно, разные. Например, на сервере валидация по заданию не нужна( точнее почти не нужна), только на gateway, поэтому и bad request, тут на сервере,  вроде бы как быть не должно. И наоборот, на  gateway не должно быть ничего,  имеющего отношение к серверу (not found). Проверьте, пожалуйста, ещё раз этот момент.

KnSwee replied:
Внес правки:
Убрал оставшуюся валидацию из server (ValidationException)
Перенес в gateway (создал ErrorHandler и прописал отлов всех ошибок валидации и их обработку)

avfyodorov reviewed (commented):
Добрый вечер, Денис! 

Хорошая работа, от меня всего пара уточнений.

KnSwee committed:
share-it 3.1
Убрана вся оставшаяся валидация из server
Добавлен отлов и обработка ошибок валидации в gateway

KnSwee commented:
Внес правки по замечаниям

avfyodorov reviewed (commented):
Добрый день, Денис!

Замечаний нет.
Работа принята.
